src/main/java/cholog/supp/api/post/service/PostService.java

```diff
@@ -19,15 +29,67 @@ public class PostService {
 
     private final PostRepository postRepository;
     private final MemberStudyMapRepository memberStudyMapRepository;
-    private final StudyGroupRepository studyGroupRepository;
+    private final CommentRepository commentRepository;
 
-    public void createPost(Member member, CreatePostRequest createPostRequest) {
+    public Long createPost(Member member, CreatePostRequest createPostRequest) {
         MemberStudyMap memberStudy = memberStudyMapRepository.findByStudyGroupIdAndMemberId(
                 createPostRequest.studyId(), member.getId())
             .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 스터디 입니다."));
         StudyGroup studyGroup = memberStudy.getStudyGroup();
-        postRepository.save(
+        Post post = postRepository.save(
             new Post(member, studyGroup, createPostRequest.title(),
                 createPostRequest.description()));
+        return post.getId();
+    }
+
+    @Transactional(readOnly = true)
+    public List<PostResponse> getPostList(PostsRequest postsRequest) {
+        List<Post> allPost = postRepository.findAllByStudyIdDesc(postsRequest.studyId());
+        return allPost.stream()
+            .map(it -> new PostResponse(it.getId(), it.getTitle(), it.getCreatedDate())).toList();
+    }
+
+    public void modifyPost(Member member, ModifyPostRequest modifyPostRequest) {
+        Post post = verifyPostOwner(member, modifyPostRequest.postId());
+        post.modifyPost(new ModifyPost(modifyPostRequest.title(), modifyPostRequest.description()));
+    }
+
+    public void deletePost(Member member, Long postId) {
+        verifyPostOwner(member, postId);
+        commentRepository.deleteAllByPostId(postId);
+        postRepository.deleteById(postId);
+    }
+
+    private Post verifyPostOwner(Member member, Long postId) {
+        Post post = postRepository.findById(postId)
+            .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 질문글 입니다."));
+        if (!Validation.verifyMember(member, post.getMember().getId())) {
+            throw new IllegalArgumentException("잘못된 접근입니다.");
+        }
+        return post;
+    }
+
+    public EachPostResponse getEachPost(Member member, Long postId) {
+        Post post = postRepository.findById(postId)
+            .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 질문글 입니다."));
+        boolean isMyPost = Validation.verifyMember(member, post.getMember().getId());
+        EachPost eachPost = new EachPost(post, isMyPost);
+        List<EachComment> comments = makeComments(post.getStudy().getId(),
+            commentRepository.findAllByPostId(postId),
+            member);
```

Comment:
```suggestion
        List<EachComment> comments = makeComments(
            post.getStudy().getId(),
            commentRepository.findAllByPostId(postId),
            member
        );
```
이렇게 아예 매개변수를 아래로 내리는게 더 보기 좋은거 같은데 어떤가요??

Reply:
오 훨씬 보기 좋다